### ISSUE TITLE
Multimesh Instances Drawn Over VoxelGI

### ISSUE THREAD

**Issue by svensk007** [godotengine/godot#72928]
### Godot version

4.0rc1

### System information

Windows 11 22H2

###### This is my first time creating an issue for an official thing so please excuse my mistakes(and point them out/ask for more info if needed)

I couldn't find anybody else with this same problem on this issues page so here it is:
When an object that has a material with transparency set to `Depth Pre-Pass` it is displayed over the bounds of a VoxelGI node

VoxelGI Selected
![editor_screenshot_2023-02-08T164006](https://user-images.githubusercontent.com/83252177/217668031-31baf134-0471-4f36-93f7-98033bbecd40.png)
VoxelGI Un-Selected (less visible but still there)
![editor_screenshot_2023-02-08T164502](https://user-images.githubusercontent.com/83252177/217668324-05a4152d-39d2-463b-bd48-98c96c819041.png)
Here it is in the attached project below
![image](https://user-images.githubusercontent.com/83252177/217670549-c45ccf44-7222-40e1-8bd9-846e313d031c.png)

### Steps to reproduce

Instance an object with material that has transparency set to `Depth Pre-Pass` and see it drawn over the VoxelGI boundaries.

### Minimal reproduction project

[issue.zip](https://github.com/godotengine/godot/files/10690547/issue.zip)

**Comment by Calinou** [comment]
Sorting in a MultiMesh is performed only once for all instances. This means transparency sorting issues are expected to occur when using alpha blending on a MultiMesh's material, especially if the MultiMesh is *partially* located outside the VoxelGI gizmo's bounds.

To avoid transparency sorting issues, use alpha scissor or alpha hash instead.

**Comment by spalose** [comment]
Would this affect more than just MultiMeshes? In my project, my base terrain mesh uses SDF mask shaders in three passes to apply its various textures. Tonight I attempted to bake a NavigationMesh for testing, and I noticed that I could only see my NavigationMesh gizmo in areas where it jutted out past my terrain mesh, because the terrain mesh's materials were taking sorting priority:

![image](https://user-images.githubusercontent.com/5170787/219935420-6e19ad36-f251-42e1-a470-89cb60ff351d.png)

**Comment by Calinou** [comment]
We could add a project setting to display the physics/navigation gizmos as x-ray, similar to what you can do with the editor 3D gizmos.

cc @smix8

**Comment by Zireael07** [comment]
@Calinou Yes please it is maddening that you can't really see the collision gizmos with the geometry...

**Comment by Calinou** [comment]
> @Calinou Yes please it is maddening that you can't really see the collision gizmos with the geometry...

On the other hand, unconditionally drawing all gizmos over solid geometry can be overwhelming. This may require fading gizmos according to distance by adjusting the shader used for drawing gizmos.

**Comment by smix8** [comment]
The navigation debug already has ProjectSettings to enable x-ray.

**Comment by svensk007** [comment]
> Would this affect more than just MultiMeshes? In my project, my base terrain mesh uses SDF mask shaders in three passes to apply its various textures.

If you have the alpha set to `depth pre-pass` then yes. As mentioned [here](https://github.com/godotengine/godot/issues/72928#issuecomment-1423353166) set the alpha to something like alpha scissor. I may be completely misunderstanding the problem but here is how I did it.

```glsl
uniform float alpha_scissor_threshold = 0.5; // add this in with all the other uniforms

void fragment {
	// all the other stuff in the fragment shader
	ALPHA_SCISSOR_THRESHOLD = alpha_scissor_threshold; // This sets the alpha type to be scissor
}
```

If this isn't at all the problem just ignore it I guess but it is what I did to fix it.
